src/App.tsx

```diff
@@ -15,35 +16,112 @@
   'Garlic',
 ];
 
+const SORT_FIELD_ALPHABETICAL = 'alphabetical';
+const SORT_FIELD_LENGTH = 'length';
+
+type SortField = 'length' | 'alphabetical' | null;
+
+interface SortingOptions {
+  sortField: SortField;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: SortingOptions,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABETICAL:
+          return good1.localeCompare(good2);
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortfield] = useState<SortField>(null);
+  const [isReversed, setIsreversed] = useState(false);
```

Comment:
The function name `setIsreversed` should be corrected to `setIsReversed` to follow the camelCase naming convention.